.github/workflows/build.yaml

```diff
@@ -22,55 +22,141 @@ jobs:
-          override: true
+      - name: Install cross
+        run: cargo install cross --git https://github.com/cross-rs/cross
+
+      - name: Configure Cross.toml
+        run: |
+          cat > Cross.toml << 'EOF'
+          [build.env]
+          passthrough = [
+            "RUST_BACKTRACE",
+            "RUST_LOG",
+            "OPENSSL_DIR",
+            "OPENSSL_LIB_DIR",
+            "OPENSSL_INCLUDE_DIR",
+            "SASL2_DIR",
+            "PKG_CONFIG_PATH",
+            "PKG_CONFIG_ALLOW_CROSS"
+          ]
+
+          [target.aarch64-unknown-linux-gnu]
+          pre-build = [
+            "dpkg --add-architecture arm64",
+            "apt-get update",
+            "apt-get install -y --no-install-recommends pkg-config libssl-dev:arm64 gcc-aarch64-linux-gnu libc6-dev-arm64-cross libsasl2-dev:arm64",
+            "export OPENSSL_DIR=/usr",
+            "export OPENSSL_LIB_DIR=/usr/lib/aarch64-linux-gnu",
+            "export OPENSSL_INCLUDE_DIR=/usr/include",
+            "export SASL2_DIR=/usr",
+            "export PKG_CONFIG_PATH=/usr/lib/aarch64-linux-gnu/pkgconfig",
+            "export PKG_CONFIG_ALLOW_CROSS=1"
+          ]
 
-      - uses: actions-rs/cargo@v1
+          [target.x86_64-unknown-linux-gnu]
+          pre-build = [
+            "apt-get update",
+            "apt-get install -y --no-install-recommends pkg-config libssl-dev libsasl2-dev",
+            "export OPENSSL_DIR=/usr",
+            "export OPENSSL_LIB_DIR=/usr/lib/x86_64-linux-gnu",
+            "export OPENSSL_INCLUDE_DIR=/usr/include",
+            "export SASL2_DIR=/usr",
+            "export PKG_CONFIG_PATH=/usr/lib/x86_64-linux-gnu/pkgconfig"
+          ]
+          EOF
+
+
+      - uses: dtolnay/rust-toolchain@stable
         with:
-          use-cross: true
-          command: build
-          args: --target ${{matrix.target}}
+          targets: ${{ matrix.target }}
+
+      - name: Build with cross
+        run: cross build --target ${{ matrix.target }} --features "rdkafka-ssl,rdkafka-sasl"
 
   build-windows:
     name: Build for windows
     runs-on: windows-latest
+    env:
+      VCPKG_ROOT: ${{ github.workspace }}/vcpkg
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
-      - uses: actions-rs/toolchain@v1
+      - name: Setup vcpkg
+        shell: bash
+        run: |
+          git clone https://github.com/microsoft/vcpkg.git
+          cd vcpkg
+          ./bootstrap-vcpkg.bat
+          ./vcpkg integrate install
+
+      - name: Install dependencies
+        shell: pwsh
+        run: |
+          $env:VCPKG_ROOT/vcpkg install openssl:x64-windows-static cyrus-sasl:x64-windows-static
+          echo "OPENSSL_DIR=$env:VCPKG_ROOT/installed/x64-windows-static" >> $env:GITHUB_ENV
+          echo "OPENSSL_STATIC=1" >> $env:GITHUB_ENV
+          echo "SASL2_DIR=$env:VCPKG_ROOT/installed/x64-windows-static" >> $env:GITHUB_ENV
+          echo "SASL2_STATIC=1" >> $env:GITHUB_ENV
```

Comment:
This seems to be wrong, can we rewrite both parts with pwsh(also fix the backslash expectations)